File: framework/backend/go/router/response.go
package router

import (
	"time"

	"github.com/powerx-plugin/framework/backend/go/bootstrap"
)

// Envelope 定义统一的响应结构。
type Envelope struct {
	Success   bool      `json:"success"`
	Data      any       `json:"data,omitempty"`
	Error     *APIError `json:"error,omitempty"`
	Message   string    `json:"message,omitempty"`
	Timestamp time.Time `json:"timestamp"`
	RequestID string    `json:"request_id,omitempty"`
}

// APIError 描述错误信息。
type APIError struct {
	Code    string `json:"code"`
	Message string `json:"message"`
	Details any    `json:"details,omitempty"`
}

// RespondSuccess 输出成功响应包。
func RespondSuccess(ctx bootstrap.Context, status int, data any, message string) {
	if ctx == nil {
		return
	}
	env := Envelope{
		Success:   true,
		Data:      data,
		Message:   message,
		Timestamp: time.Now().UTC(),
		RequestID: requestIDFromContext(ctx),
	}
	ctx.JSON(status, env)
}

// RespondError 输出带错误信息的响应包。
func RespondError(ctx bootstrap.Context, status int, code, message string, details any) {
	if ctx == nil {
		return
	}
	env := Envelope{
		Success:   false,
		Error:     &APIError{Code: code, Message: message, Details: details},
		Timestamp: time.Now().UTC(),
		RequestID: requestIDFromContext(ctx),
	}
	ctx.JSON(status, env)
}

func requestIDFromContext(ctx bootstrap.Context) string {
	if ctx == nil {
		return ""
	}
	if id := ctx.Header("X-Request-ID"); id != "" {
		return id
	}
	if id := ctx.Header("Request-ID"); id != "" {
		return id
	}
	return ""
}

File: skeleton/backend/python-fastapi/app/contracts/response.py
from dataclasses import dataclass
from datetime import datetime
from typing import Any, Optional


@dataclass
class APIError:
    code: str
    message: str
    details: Optional[dict[str, Any]] = None


@dataclass
class APIResponse:
    success: bool
    message: str
    data: Any
    error: Optional[APIError]
    timestamp: str
    request_id: Optional[str]


def ok(data: Any = None, message: str = "") -> APIResponse:
    return APIResponse(
        success=True,
        message=message,
        data=data,
        error=None,
        timestamp=datetime.utcnow().isoformat() + "Z",
        request_id=None,
    )


def fail(code: str, message: str, details: Optional[dict[str, Any]] = None) -> APIResponse:
    return APIResponse(
        success=False,
        message=message,
        data=None,
        error=APIError(code=code, message=message, details=details),
        timestamp=datetime.utcnow().isoformat() + "Z",
        request_id=None,
    )
